code/modules/projectiles/guns/pistols.dm

```diff
@@ -76,6 +76,51 @@
 	recoil_unwielded = -2
 
 //-------------------------------------------------------
+//TX-7 Plasma Pistol
+/obj/item/weapon/gun/pistol/plasma_pistol
+	name = "\improper TX-7 plasma pistol"
+	desc = "An experimental weapon. Designed to set the terrain it is fired at on fire so as to remove flammable obstacles and clear out dug in enemies from some range."
```

Comment:
Cool guns need cool descriptions. (You don't have to use it.)

"An experimental weapon designed to set the terrain and targets on fire. It hums whit power as magnetic fields coil round each other."

Reply:
Thanks for the suggestion, I used it. : )